public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx

```diff
@@ -157,48 +135,25 @@ export const Filter = ({ columns, rawData, setFilteredData }: FilterProps) => {
                 onClick={() => onClearAll(form)}
                 data-testid="clear-all-button"
               />
+              {hasBackendFiltering && (
+                <IconButton className={styles.icon} name="check" size="xl" type="submit" data-testid="submit-button" />
+              )}
             </div>
           </div>
           {showAdvanceFilter && openCollapse && (
             <div className={styles.advanceFilter}>
               {columns.map((column) => {
-                const columnOptions = buildColumnOptions(column);
                 if (column.type === FilterFieldTypes.DROPDOWN) {
-                  return (
-                    <div>
-                      <Field name={`${column.accessor}`}>
-                        {({ input }) => (
-                          <SelectField
-                            options={columnOptions}
-                            defaultValue={{ value: ALL_VALUE, label: ALL_LABEL }}
-                            label={column.label ?? column.Header}
-                            {...input}
-                            data-testid="select-dropdown"
-                          />
-                        )}
-                      </Field>
-                    </div>
-                  );
+                  return <SelectDropdownField column={column} />;
                 }
                 if (column.type === FilterFieldTypes.RADIO_BUTTON) {
-                  return (
-                    <div>
-                      <RadioButtonGroupField
-                        options={columnOptions}
-                        defaultValue={ALL_VALUE}
-                        name={`${column.accessor}`}
-                        label={column.label ?? column.Header}
-                        fullWidth
-                        data-testid="radio-button"
-                      />
-                    </div>
-                  );
+                  return <RadioButtonField column={column} />;
```

Comment:
I think we can just do `column.type === FilterFieldTypes.RADIO_BUTTON && <RadioButtonField column={column} />`

Same for the other

Reply:
Cool, I didn't know that it is possible to do it like this 👍